### ISSUE TITLE
Forms: Discuss field and label IDs

### ISSUE THREAD

**Issue by mparke** [linode/manager#1007]
Discuss and review input ids and for attributes. Apply `id`s to fields and `for` attributes to labels where appropriate.

**Comment by mparke** [comment]
some notes added here: https://github.com/linode/manager/wiki/Development-FAQ#when-to-use-classes-or-ids

since we've noted that ids should be used to link inputs and labels, I feel we can probably mark this done. 

will wait for another thumbsup

**Comment by eatonphil** [comment]
The part of this I most wanted to discuss, personally, was whether or not it's acceptable to auto-generate these ids.

**Comment by mparke** [comment]
almost forgot, thoughts on auto generating ids?

**Comment by eatonphil** [comment]
One solution (anything along these lines):

```javascript
render() {
  const ids = generateUniqueIds(30);

  return (
    <form>
      <div>
        <label id={withPrefix(ids[0], 'myinput') />
        <input id={withPrefix(ids[0], 'myinput'} />
      </div>
      <div>
        <label id={withPrefix(ids[1], 'myinput') />
        <input id={withPrefix(ids[1], 'myinput'} />
      </div>
    </form>
  )
}
```

**Comment by mparke** [comment]
thought about this for a little while, and I'm not sure if there's an obvious benefit to having a unique id counter.

nonetheless, some options:

**my preferred choice:**
no id counter or prefixes, keep it simple, be aware of what forms are on the page
```
// ...form module
return (
  <FormGroup>
    <label for="name"></label>
    <input id="name"/>
  </FormGroup>
) 
```

as part of a module/form:

```
import { globalIdGenerator } from '~/globalIdGenerator';

export default function myform() {
  const nameId = globalIdCounter('name');

  // use in field/label
};
```

as part of form group

```
let globalIdCounter = 0;

export default function FormGroup(props) {
  const id = `${props.name}-${++globalIdCounter}`;
  
  return (
    <div className="form-group">
      {props.label(id, ...)}
      {props.field(id, ...)}
    </div>  
  )
}
```

**Comment by mparke** [comment]
any thoughts @abemassry @na3d  ?

**Comment by eatonphil** [comment]
> no id counter or prefixes, keep it simple, be aware of what forms are on the page

I'm fine with this until we know we need something more / less.

**Comment by abemassry** [comment]
I would say go with your preferred choice and if that is not sufficient then have a counter associated with each one like in your examples.  Naming things is hard but not impossible, I've generated some pretty unique names, the only thing that ends up hindering me is the maximum line length put forward by the linter.

That being said there are times when I had to come up with a system for auto generating unique ids and I did something very similar because the user was generating an arbitrary number of dom elements and they had to be uniquely tracked and accounted for, and I didn't want to have to ask them for a unique name for each element they created so I came up with a similar system.  I guess thats the only use case I could think of for needing a system like this.

TLDR: I'd go with @mparke on this.

**Comment by mparke** [comment]
General usage has consensus, closing.
